MinimedKit/Models/PumpModel.swift

```diff
@@ -134,6 +127,27 @@ public enum PumpModel: String {
     public var minimumBasalScheduleEntryDuration: TimeInterval {
         return .minutes(30)
     }
+
+    public var variableDeliveryRate: Bool {
+        return generation >= 23
+    }
+
+    public func bolusDeliveryTime(units: Double) -> TimeInterval {
+        let unitsPerMinute: Double
+        if variableDeliveryRate {
+            switch units {
+            case let u where u < 1.0:
+                unitsPerMinute = 0.75
+            case let u where u > 7.5:
+                unitsPerMinute = units / 5
```

Comment:
Is this really `units / 5`? Meaning that if I'm bolusing 7.7 it will deliver 1.54 units per minute? Just curious.

Reply:
Yes, on average, over that 5 minutes. In reality there are three different rates, each pulsed once every 2 seconds. Fastest at the beginning, then slowing down, all adding up to the amount of delivery over 5 minutes.